tests/fixture/bootstrapmonitor/wait.go

```diff
@@ -0,0 +1,194 @@
+			log.Error("failed to check node health", zap.Error(err))
+			return false, nil
+		} else {
+			if !healthy.Healthy {
+				log.Info("Node reported unhealthy", commonFields...)
+				return false, nil
+			}
+
+			log.Info("Node reported healthy")
+		}
+
+		commonFields = append(commonFields, zap.Reflect("testConfig", testConfig))
+		log.Info("Bootstrap completed successfully", commonFields...)
+
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("failed to wait for node to report healthy: %w", err)
+	}
+
+	log.Info("Waiting for new image to test")
+	if err := wait.PollImmediateInfinite(imageCheckInterval, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), contextDuration)
+		defer cancel()
+
+		log.Info("Starting pod to get the image id for the `latest` tag")
+		latestImageDetails, err := getLatestImageDetails(ctx, log, clientset, namespace, testConfig.Image, nodeContainerName)
+		if err != nil {
+			log.Error("failed to get latest image id", zap.Error(err))
+			return false, nil
+		}
+
+		if latestImageDetails.Image == testConfig.Image {
+			log.Info(ImageUnchanged)
+			return false, nil
+		}
+
+		log.Info("Found updated image",
+			zap.String("image", latestImageDetails.Image),
+			zap.Reflect("versions", latestImageDetails.Versions),
+		)
+
+		log.Info("Updating StatefulSet to trigger a new test")
+		if err := setImageDetails(ctx, log, clientset, namespace, podName, latestImageDetails); err != nil {
+			log.Error("failed to set container image", zap.Error(err))
+			return false, nil
+		}
+
+		// Statefulset will restart the pod with the new image
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("failed to wait for new image to test: %w", err)
+	}
+
+	// Avoid exiting immediately to avoid container restart before the pod is recreated with the new image
+	time.Sleep(5 * time.Minute)
+	return nil
+}
+
+// Determines the current disk usage for the specified directory
+func getDiskUsage(log logging.Logger, dir string) string {
+	cmd := exec.Command("du", "-sh", dir)
+
+	// Create a buffer to capture stderr in case an unexpected error occurs
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+
+	output, err := cmd.Output()
+	if err != nil {
+		exitError, ok := err.(*exec.ExitError)
+		if !ok {
+			log.Error("Error executing du", zap.Error(err))
+			return ""
+		}
+		switch exitError.ExitCode() {
+		case 1:
+			// Exit code 1 usually indicates that files cannot be accessed. Since avalanchego will
+			// regularly delete files in the db dir, this can be safely ignored and the regular disk
+			// usage message can be printed.
+		case 2:
+			log.Error("Incorrect usage of du command for dir",
+				zap.String("dir", dir),
+				zap.String("stderr", stderr.String()),
+			)
+			return ""
+		default:
+			log.Error("du command failed for dir",
+				zap.Int("exitCode", exitError.ExitCode()),
+				zap.String("dir", dir),
+			)
+			return ""
+		}
```

Comment:
Should we just append the original `exitError` to these logs for these cases anyways?